store/sqlstore/status_store.go

```diff
@@ -99,6 +101,38 @@ func (s SqlStatusStore) GetByIds(userIds []string) ([]*model.Status, error) {
 	return statuses, nil
 }
 
+func (s SqlStatusStore) UpdateExpiredDNDStatuses() ([]*model.Status, error) {
+	queryString := fmt.Sprintf(`
+		UPDATE
+			Status
+		SET
+			DNDEndTimeUnix = %v, Manual = %v,
+			PrevStatus = '%v', Status = PrevStatus
+		WHERE
+			(Status = '%v' AND DNDEndTimeUnix <= %v AND DNDEndTimeUnix > %v)
```

Comment:
We can experiment a bit here when we start load-testing this feature with the indexes. Creating a new index around DNDEndTimeUnix can speedup drastically this query, but at the same time can add a bit more load on the status writes. We can test it.